fsspec/implementations/memory.py

```diff
@@ -114,6 +114,15 @@ def makedirs(self, path, exist_ok=False):
             if not exist_ok:
                 raise
 
+    def pipe_file(self, path, value, **kwargs):
+        """Set the bytes of given file
+
+        Avoids copies of the data if possible
+        """
+        with self.open(path, "wb", data=value, **kwargs) as f:
+            # value passed into initializer to avoid copy
+            assert f.getvalue() is value
```

Comment:
An `assert` in non-test code seems wrong to me - we know this should always work. Since this is within our own control, should we not just `self.store[path] = MemoryFile(data)` or `self.open(path, "wb", data=value, **kwargs)` (no context)? I don't actually think there are any valid kwargs here.

Reply:
Agreed.  ..and while I don't think there are an valid kwargs here, should we leave the `**kwargs` in place, on the potential of future changes?

Reply:
Followed recommendations here in this conversation, and both changes have been made.